### DISCUSSION TITLE
Appium inspector installs and uninstalls the webdriveragent and doesn't allow to run apps on real iOS device

### DISCUSSION THREAD

**Discussion by vamc264** [appium/appium#19600] [Q&A] [answered]
I built and installed Webdrivergent using Xcode from `/Users/$USER/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent` without any issues from the documentation: https://github.com/appium/appium-xcuitest-driver/blob/master/docs/real-device-config.md#appium-2x

**Appium Inspector Versions Tried**:
App Version: 2023.12.2
App Version: 2023.10.4

**Used Versions:**
**Appium** - v2.3.0 & v2.0.1
**Node** - v20.10.0
**npm** -v10.2.3
**Appium Location**: /usr/local/bin/appium
**Carthage Location**: /usr/local/bin/carthage
**MacOS: Sonoma**- 14.1.2
**Java Version**:
`java 17.0.2 2022-01-18 LTS
Java(TM) SE Runtime Environment (build 17.0.2+8-LTS-86)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.2+8-LTS-86, mixed mode, sharing)`

**Devices Used:**
iPhone 6s: OS- 15.8
iPhone 12: OS 16.5

**Appium doctor** - v1.16.0 and info below
<img width="1195" alt="Screenshot 2024-01-03 at 15 21 40" src="https://github.com/appium/appium/assets/24605723/52110b30-671b-44d3-b74e-e827bf196d08">

**Appium Drivers list**
<img width="395" alt="Screenshot 2024-01-03 at 15 26 53" src="https://github.com/appium/appium/assets/24605723/52d3d01a-1662-44c7-9985-a659c18c9153">

**Here is the error shown on the appium inspector**
<img width="363" alt="Screenshot 2024-01-03 at 15 13 03" src="https://github.com/appium/appium/assets/24605723/bcbf0118-db21-4e95-b659-031d097fa633">

**Capabilities Used:**
<img width="459" alt="Screenshot 2024-01-03 at 15 15 23" src="https://github.com/appium/appium/assets/24605723/485c2ba3-f39f-4194-9d10-da8507df83e4">

**Terminal Appium Logs:**
<img width="1278" alt="Screenshot 2024-01-03 at 15 43 53" src="https://github.com/appium/appium/assets/24605723/c5f6b5ba-1883-44ce-a892-03f3bb7eb7a4">
<img width="1276" alt="Screenshot 2024-01-03 at 15 42 57" src="https://github.com/appium/appium/assets/24605723/475ddfab-12c2-40a4-98cd-ddf1135ea45d">
<img width="1278" alt="Screenshot 2024-01-03 at 15 41 44" src="https://github.com/appium/appium/assets/24605723/f442c23d-89f3-4126-a87a-20808d20f0df">

I've uninstalled and reinstalled both Appium and Xcode. Experimented with various versions of both, but I'm uncertain about the root cause of the issue. I explored alternative iOS inspectors without success and reviewed help from various sources, but the problem persists. If you need logs or any specific information, please feel free to request them, and I'd appreciate assistance in resolving this matter.
 
 Thanks in Advance.

**Comment by jlipps** [comment]
Please paste your Xcode log as a gist. Screenshots are not very helpful. Also `appium:USE_NEW_WDA` is not going to do anything. It's not a recognized capability. Xcode error 65 is usually a signing issue during build.

**Reply by vamc264** [reply]
Hi @jlipps, Thank you for your quick reply. I have removed the `appium:USE_NEW_WDA` cap. To my understanding, if there were a signing issue, Xcode should have displayed an error message during the build or testing of WebdriverAgent. However, the process was successful. Please correct me if there are any mistakes in how I signed the build. By the way, the Apple account used is self-owned. Here are the logs: https://gist.github.com/vamc264/d8a584bffef8a5cea21c6acb2daa7afc

Feel free to inform me if you require any additional information.

**Reply by jlipps** [reply]
what happens when you run the command that appium is running under the hood? 

```
/Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild test-without-building -project /Users/work/.appium/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -derivedDataPath /Users/work/Library/Developer/Xcode/DerivedData/WebDriverAgent-cbilvmcmehvjbvfmfkgspqgqrxxm -destination id=0877056afa83043ae7d307187c2f7484305b3442 IPHONEOS_DEPLOYMENT_TARGET=15.8 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO
```

**Reply by vamc264** [reply]
@jlipps I see the WebDriverAgent is installed and running on the device.  Below is the log, when I run the above command:

https://gist.github.com/vamc264/8cf87af405e0e5348ca2320fa872fa0d

**Reply by vamc264** [reply]
@jlipps I attempted to utilize a different Apple ID and assigned a distinct bundle ID for the WebdriverAgent project. The installation of WebdriverAgent was successful when testing via Xcode. However, when using Appium Inspector, it still failed with the same error as mentioned earlier. I'm unsure about the root cause of the issue. Any additional suggestions or thoughts I should explore?

**Reply by KazuCocoa** [reply]
Not sure the entire behavior as no full Appium log, but error code 65 by xcodebuild mostly means a signing issue.
Xcode and xcodebuild might refer to different debug build caches. (`DerivedData` path location). Then, the xcodebuild's one refers to the prebuilt, you may need to re-build the prebuilt module again. Your config specified `usePrebuiltWDA`, then the Appium process won't re-build the module again. (not calls `build-for-testing`). It could keep using the old (wrong?) sign's build for example. I guess potentially `appium:allowProvisioningDeviceRegistration` caps help if your certificate hasn't registered the device yet in the re-building.

https://appium.github.io/appium-xcuitest-driver/5.12/run-prebuilt-wda/ might help since your caps specified `usePrebuiltWDA`.

Btw, we observed free account could fail to launch app via xcodebuild in the past. The behavior depends on Apple, but it could cause similar behavior. (not vis Xcode. via xcodebuild command behavior.)

**Reply by burtonemily** [reply]
I had the same issue with error code 65 and i needed to update WebDriverAgent and reinstall Appium. 
`appium driver update xcuitest` 
My problem then was the importing of W3CCapabilities from my appium folder, so then I changed the import from 
`import { W3CCapabilities } from "appium/build/lib/appium";` to `import { W3CCapabilities } from "@wdio/types/build/Capabilities";`

Now I'm going through my code and updating the typings because I think the new webdriver update has a bunch of typing that were previously not included. 

Not sure if this will help you or fix your problem but its on the way to fixing mine. 

TLDR; it wasn't an XCode issue it was a WebDriverAgent issue (it needs to update)

**Reply by vamc264** [reply]
@burtonemily I have tried updating the WebDriverAgent using `appium driver update xcuitest` and I see the below error.


```
✔ Checking if driver 'xcuitest' is updatable
✔ Checking if driver 'xcuitest' needs an update
⠋ Updating driver 'xcuitest' from 5.12.2 to 5.13.2⠋ Installing 'xcuitest' using NPM install spec 'appium-✖ Installing 'xcuitest' using NPM install spec 'appium-xcuitest-driver@5.13.2'
✖ Updating driver 'xcuitest' from 5.12.2 to 5.13.2
ℹ Update report:
✖ - 'xcuitest' failed to update: Error: ✖ Encountered an error when installing package: npm command 'install --save-dev --no-progress --no-audit --prefer-offline --omit=peer --save-exact --global-style --no-package-lock appium-xcuitest-driver@5.13.2 --json' failed with code 1.

STDOUT:
{
"error": {
"code": "ETARGET",
"summary": "No matching version found for appium-xcuitest-driver@5.13.2.",
"detail": "In most cases you or one of your dependencies are requesting\na package version that doesn't exist."
}
}

STDERR:
npm WARN config global-style This option has been deprecated in favor of --install-strategy=shallow
npm ERR! code ETARGET
npm ERR! notarget No matching version found for appium-xcuitest-driver@5.13.2.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```

<img width="740" alt="Screenshot 2024-01-12 at 10 26 06" src="https://github.com/appium/appium/assets/24605723/91880595-80a3-4b5c-92e2-5613c05e9ef4">

**Reply by KazuCocoa** [reply]
> No matching version found for appium-xcuitest-driver@5.13.2.

This was npm cache issue. Please use  the latest appium that has a fix for the npm installation command. Or you could run `npm cache clean --force` run once to clear your local npm cache.

**Comment by vamc264** [comment, marked as answer]
Hi, Webdriveragent is running on a real device now. It has been solved by uninstalling the appium and re-installing the appium(`npm install -g appium` )with the xcuitest driver version from @5.12.2 to @5.14.0 (`appium driver update xcuitest`) 

Thank you everyone for your help and time in looking into this issue. 

closing this ticket as it got solved.
